libmamba/include/mamba/core/util_url.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_CORE_UTIL_URL_HPP
+#define MAMBA_CORE_UTIL_URL_HPP
+
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace mamba
+{
+    std::string concat_scheme_url(const std::string& scheme, const std::string& location);
+    std::string build_url(
+        const std::optional<std::string>& auth,
+        const std::string& scheme,
+        const std::string& base,
+        bool with_credential
+    );
+    void split_platform(
+        const std::vector<std::string>& known_platforms,
+        const std::string& url,
+        const std::string& context_platform,
+        std::string& cleaned_url,
+        std::string& platform
+    );
```

Comment:
Yes they are only used in the channels but since they're independent functions and could be used elsewhere, I put them in a kind of utility file to be available and not reimplemented if needed. If we don't want them to be publically available, we could also not put the `util_url.hpp` as a public header (`url.hpp` is).
It's like `fix_win_path` which is used only in the channels but could be in `util_os` instead (I moved it there as well).
What do you think? Do you prefer leaving them where they were (in the channels)? (by the way I was thinking about doing the same about some independent functions located in `url.hpp` but in this case they are used elsewhere though).

Reply:
Actually they wil probably be used elsewhere, so having them in a header is fine. For consistency I think they shouldbe moved to the URL header (we already have a bunch of split functions there).

Reply:
Having everything in url would make it easier to find (the other `util_xxx.hpp` do not have a `xxx.hpp` standalone counterpart).